pkg/controller/registry/grpc/source.go

```diff
@@ -228,6 +228,11 @@ func (s *SourceStore) watch(ctx context.Context, key registry.CatalogKey, source
 					s.sources[key] = *src
 					s.sourcesLock.Unlock()
 
+					// always try to reconnect if the connection goes idle
+					if newState == connectivity.Idle {
+						source.Conn.Connect()
```

Comment:
Looks like this already no-ops if the underlying connections are not idle, so it might be better to call this unconditionally. A comment to note that this is non-blocking would be useful as well as it impacts the understanding of what consumers of `s.notify` see.

Reply:
Trying this in #3144, which should bring OLM to the minimal version of GRPC necessary for the next set of updates